apis/core/v1alpha1/function_types.go

```diff
@@ -24,97 +24,94 @@ import (
 	SourceSubPath *string `json:"sourceSubPath,omitempty"`
-	// Clean out the contents of the repo's destination directory if it already exists before cloning the repo there (default: true)
-	DeleteExisting *bool `json:"deleteExisting,omitempty"`
-	// Git HTTP proxy server for non-SSL requests (default: "")
-	HttpProxy *string `json:"httpProxy,omitempty"`
-	// Git HTTPS proxy server for SSL requests (default: "")
-	HttpsProxy *string `json:"httpsProxy,omitempty"`
-	// Git no proxy - opt out of proxying HTTP/HTTPS requests (default: "")
-	NoProxy *string `json:"noProxy,omitempty"`
-	// Log the commands that are executed during git-clone's operation (default: true)
-	Verbose *bool `json:"verbose,omitempty"`
-	// The image used where the git-init binary is (default: "gcr.io/tekton-releases/github.com/tektoncd/pipeline/cmd/git-init:v0.17.3")
-	GitInitImage *string `json:"gitInitImage,omitempty"`
+	// Credentials references a Secret that contains credentials to access
+	// the repository.
+	//
+	// +optional
+	Credentials *v1.LocalObjectReference `json:"credentials,omitempty"`
 }
 
 func (gr *GitRepo) Init() {
-	var revision, refspec, subDir, sourceSubPath, httpProxy, httpsProxy, noProxy, gitInitImage string
-	var deletingExisting, submodules, sslVerify, verbose bool
-	var depth int8
+	var revision, sourceSubPath string
 	gr.Revision = &revision
-	gr.Refspec = &refspec
-	gr.SubDirectory = &subDir
 	gr.SourceSubPath = &sourceSubPath
-	gr.DeleteExisting = &deletingExisting
-	gr.HttpProxy = &httpProxy
-	gr.HttpsProxy = &httpsProxy
-	gr.NoProxy = &noProxy
-	gr.GitInitImage = &gitInitImage
-	gr.Submodules = &submodules
-	gr.SslVerify = &sslVerify
-	gr.Verbose = &verbose
-	gr.Depth = &depth
-}
-
-type Registry struct {
-	// Image registry url
-	Url *string `json:"url,omitempty"`
-	// Image registry account including username and password
-	Account *v1.SecretKeySelector `json:"account,omitempty"`
-}
-
-func (r *Registry) Init() {
-	var url string
-	r.Url = &url
-	r.Account = &v1.SecretKeySelector{}
+	gr.Credentials = &v1.LocalObjectReference{}
 }
 
 type Language string
 type Runtime string
 
 const (
 	Go            Language = "go"
-	Python        Language = "python"
 	Node          Language = "node"
 	BuildPhase             = "Build"
 	ServingPhase           = "Serving"
 	Created                = "Created"
-	Launching              = "Launching"
-	Launched               = "Launched"
+	Building               = "building"
+	Running                = "Running"
+	Succeeded              = "Succeeded"
 	Failed                 = "Failed"
+	Skipped                = "Skipped"
 	Knative       Runtime  = "Knative"
 	OpenFuncAsync Runtime  = "OpenFuncAsync"
+	Shipwright             = "Shipwright"
 )
 
+type Strategy struct {
+	// Name of the referent; More info: http://kubernetes.io/docs/user-guide/identifiers#names
+	Name string `json:"name"`
+
+	// BuildStrategyKind indicates the kind of the buildstrategy, BuildStrategy or ClusterBuildStrategy, default is BuildStrategy.
```

Comment:
// BuildStrategyKind indicates the kind of the build strategy: BuildStrategy or ClusterBuildStrategy, default to BuildStrategy.